File: src/api/java/Z3Exception.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    Z3Exception.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/ 

package com.microsoft.z3;


/**
 * The exception base class for error reporting from Z3
 **/
@SuppressWarnings("serial")
public class Z3Exception extends Exception
{
    /**
     * Constructor.
     **/
    public Z3Exception()
    {
        super();
    }

    /**
     * Constructor.
     **/
    public Z3Exception(String message)
    {
        super(message);
    }

    /**
     * Constructor.
     **/
    public Z3Exception(String message, Exception inner)
    {
        super(message, inner);
    }
}

File: src/api/dotnet/Z3Exception.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    Exception.cs

Abstract:

    Z3 Managed API: Exceptions

Author:

    Christoph Wintersteiger (cwinter) 2012-03-15

Notes:

--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// The exception base class for error reporting from Z3
    /// </summary>
#if !DOTNET_CORE
  [Serializable]
#endif
    public class Z3Exception : Exception
  {
    /// <summary>
    /// Constructor.
    /// </summary>
    public Z3Exception() : base() { }

    /// <summary>
    /// Constructor.
    /// </summary>
    public Z3Exception(string message) : base(message) { }

    /// <summary>
    /// Constructor.
    /// </summary>
    public Z3Exception(string message, System.Exception inner) : base(message, inner) { }
  }
}
